debian/ellis.init.d

```diff
@@ -68,6 +68,31 @@ SCRIPTNAME=/etc/init.d/$NAME
 # Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
 . /lib/lsb/init-functions
 
+# Check if any of Ellis's tools are currently running.
+tools_running()
+{
+  pgrep -f "src/metaswitch/ellis/tools" >/dev/null 2>&1
+}
+
+# If any of Ellis's tools are currently running wait up to 20s for them to
+# complete.
+wait_for_tools()
+{
+  if tools_running; then
+    log_daemon_msg "Waiting for tools to complete..." "$NAME"
+
+    for i in `seq 1 20`; do
+      if tools_running; then
+        sleep 1
+      else
+        break
+      fi
+    done
+
+    log_daemon_msg "Tools finished" "$NAME"
```

Comment:
Agreed. I am going to output a different log if we gave up, and also change the init script to return an error if this is the case.